android/app/src/main/cpp/app/organicmaps/util/SpeedFormatted.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "app/organicmaps/core/jni_helper.hpp"
+
+#include "platform/speed_formatted.hpp"
+
+inline jobject ToJavaSpeedFormatted(JNIEnv * env, platform::SpeedFormatted const & speedFormatted)
+{
+  static jclass const speedFormattedClass = jni::GetGlobalClassRef(env, "app/organicmaps/util/SpeedFormatted");
+
+  static jmethodID const speedFormattedConstructor = jni::GetConstructorID(env, speedFormattedClass, "(DLjava/lang/String;B)V");
+
+  jobject distanceObject = env->NewObject(
+      speedFormattedClass, speedFormattedConstructor,
+      speedFormatted.GetSpeed(), jni::ToJavaString(env, speedFormatted.GetSpeedString()), static_cast<uint8_t>(speedFormatted.GetUnits()));
```

Comment:
nit: Looks like a lot of code, with a lot of jni overhead. Can formatting/string preparation be done all in C++ code, passed as several strings (or maybe just one string returned as a concatenation of several strings?) to the iOS and Android UI just for displaying?